src/status_im/contexts/profile/contact/block_contact/view.cljs

```diff
@@ -0,0 +1,70 @@
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn on-close [] (rf/dispatch [:hide-bottom-sheet]))
+
+(defn view
+  []
+  (let [{:keys [customization-color contact-request-state public-key]
+         :as   contact}                       (rf/sub [:contacts/current-contact])
+        ;; TODO(@mohsen): remove :blue, https://github.com/status-im/status-mobile/issues/18733
+        customization-color                   (or customization-color :blue)
+        full-name                             (profile.utils/displayed-name contact)
+        profile-picture                       (profile.utils/photo contact)
+        [remove-contact? set-remove-contact?] (rn/use-state false)
+        on-remove-change                      (rn/use-callback #(set-remove-contact? %))
+        on-block-press                        (rn/use-callback
+                                               (fn []
+                                                 (rf/dispatch [:toasts/upsert
+                                                               {:id   :user-blocked
+                                                                :type :positive
+                                                                :text (i18n/label :t/user-blocked
+                                                                                  {:username
+                                                                                   full-name})}])
+                                                 (rf/dispatch [:contact.ui/block-contact-confirmed
+                                                               public-key])
+                                                 (when remove-contact?
+                                                   (rf/dispatch [:contact.ui/remove-contact-pressed
+                                                                 {:public-key public-key}]))
+                                                 (on-close))
+                                               [remove-contact? public-key full-name])]
+    [:<>
+     [quo/drawer-top
+      {:type                :context-tag
+       :context-tag-type    :default
+       :title               (i18n/label :t/block)
+       :full-name           full-name
+       :profile-picture     profile-picture
+       :customization-color customization-color}]
+     [rn/view {:style style/content-wrapper}
+      [quo/text
+       {:weight :medium
+        :size   :paragraph-1}
+       (i18n/label :t/block-user-title-message {:username full-name})]
+      [quo/information-box
+       {:icon :i/info
+        :type :default}
+       (i18n/label :t/blocking-a-user-message {:username full-name})]
+      (when (= constants/contact-request-state-mutual contact-request-state)
+        [rn/pressable
+         {:style    style/checkbox-wrapper
+          :on-press #(on-remove-change (not remove-contact?))}
```

Comment:
Should this also be a hook?

Reply:
wdym? can you give me an example? `on-remove-change` is a `use-callback`

Reply:
I think when we wrap `on-remove-change` with an anonymous function `#(on-remove-change)` it will pass a different function reference each time it's rendered. But maybe we can avoid this by using something like:
```
(use-callback (fn [] (on-remove-change (not remove-contact?))) [remove-contact?])
```
though there may be another way to memoize a partially applied function with `use-memo` and `partial` 🤔

Reply:
I prefer to avoid creating multiple hook for one set state, i think here 
`on-remove-toggle                      (rn/use-callback #(set-remove-contact? not) [])` 
is a better option